### PR TITLE
JDK-8301205: Port fdlibm log10 to Java

### PR DESCRIPTION
**Pull request by jddarcy**
Restarting the port of FDLIBM to Java with the log10 method.

There are two port, the first a near-transliteration from C port to use as a test reference in

test/jdk/java/lang/StrictMath/FdlibmTranslit.java

and a more idiomatic Java port in 

src/java.base/share/classes/java/lang/FdLibm.java

First I debug the FdlibmTranslit port by running it against the C version. Once that is working, I copy the port into the java.lang.FdLibm.java and do a series of transformation to get the code closer to usual JDK style.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301205](https://bugs.openjdk.org/browse/JDK-8301205): Port fdlibm log10 to Java

### Reviewers
 * [Brian Burkhalter](https://openjdk.org/census#bpb) (@bplb - **Reviewer**) ⚠️ Review applies to [0f74dd5f](https://git.openjdk.org/jdk/pull/12242/files/0f74dd5f88f8ff1f4e97674b53becf4e9292f74e)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12242/head:pull/12242` \
`$ git checkout pull/12242`

Update a local copy of the PR: \
`$ git checkout pull/12242` \
`$ git pull https://git.openjdk.org/jdk pull/12242/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12242`

View PR using the GUI difftool: \
`$ git pr show -t 12242`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12242.diff">https://git.openjdk.org/jdk/pull/12242.diff</a>

</details>

### CODE REVIEW

**Comment by jddarcy** [comment]
PS The C file used as the basis of the port is

src/java.base/share/native/libfdlibm/e_log10.c

One difference to highlight between the original C and transliteration port is the line

i  = ((unsigned)k&0x80000000)>>31; // C version
vs
i  = (k&0x80000000)>>>31; // unsigned shift

File: src/java.base/share/classes/java/lang/FdLibm.java [hunk 1]
```diff
@@ -745,4 +745,81 @@ public static double compute(double x) {
...
+
+            if (hx >= 0x7ff0_0000) {
+                return x + x;
+            }
+
+            k += (hx >> 20) - 1023;
+            i  = (k  & 0x8000_0000) >>> 31; // unsigned shift
+            hx = (hx & 0x000f_ffff) | ((0x3ff - i) << 20);
+            y  = (double)(k + i);
+            x = __HI(x, hx);
```
Comment (rgiulietti) [inline src/java.base/share/classes/java/lang/FdLibm.java:820]: `            x = __HI(x, hx);  // replace high word of x with hx`

Comment (rgiulietti) [reply src/java.base/share/classes/java/lang/FdLibm.java:820]: Otherwise LGTM

Comment (jddarcy) [reply src/java.base/share/classes/java/lang/FdLibm.java]: Okay; comment added back.

**Review by bplb — APPROVED** [review]
Looks fine.

**Pull request closed without merging**